plutus-benchmark/marlowe/src/PlutusBenchmark/Marlowe/Core/V1/Semantics.hs

```diff
@@ -322,19 +322,20 @@ evalObservation env state obs = let
 
 -- | Pick the first account with money in it.
 refundOne :: Accounts -> Maybe ((Party, Token, Integer), Accounts)
-refundOne accounts = case Map.toList accounts of
-    [] -> Nothing
-    -- SCP-5126: The return value of this function differs from
-    -- Isabelle semantics in that it returns the least-recently
-    -- added account-token combination rather than the first
-    -- lexicographically ordered one. Also, the sequence
-    -- `Map.fromList . tail . Map.toList` preserves the
-    -- invariants of order and non-duplication.
-    ((accId, token), balance) : rest ->
-        if balance > 0
-        then Just ((accId, token, balance), Map.fromList rest)
-        else refundOne (Map.fromList rest)
-
+refundOne accounts =
+    if Map.null accounts
+        then Nothing
+        else
+            -- SCP-5126: The return value of this function differs from
+            -- Isabelle semantics in that it returns the least-recently
+            -- added account-token combination rather than the first
+            -- lexicographically ordered one. Also, the sequence
+            -- `Map.fromList . tail . Map.toList` preserves the
+            -- invariants of order and non-duplication.
+            let (((accId, token), balance), rest) = Map.unsafeUncons accounts
```

Comment:
Is the comment still true? At the very least the `Map.fromList . tail . Map.toList point seems obsolete. Someone from the Marlowe team should look at this, I think.